src/clj/muuntaja/parse.clj

```diff
@@ -7,16 +7,18 @@
 ;;
 
 (defn fast-memoize [size f]
-  (let [cache (ConcurrentHashMap. size 0.8 4)]
+  (let [cache (ConcurrentHashMap. size 0.8 4)
+        empty '(nil)]
     (fn [& args]
       (or (.get cache args)
-          (let [ret (apply f args)]
-            (when ret
-              ;; not synchronized but ok enough
-              (when (> (.size cache) size)
-                (.clear cache))
-              (.putIfAbsent cache args ret))
-            ret)))))
+          (if-not (= args empty)
```

Comment:
Not sure what this is? Always return nil if for `((fast-memoize .. f) nil)`?

Reply:
good catch. we need to either guard against input to the memoize or handle the nil-case within it. Without this, the `((fast-memoize .. f) nil)` always misses the cache (`'(nil)` is not found on the cache as it returns `nil` and the `.get` returns `nil`. Not sure why. But the execution cost goes from ~30ns to 3000ns. For cases like "no accept-header set" etc.

Reply:
How about using `.getOrDefault`?

```clojure
(let [cache ...
      sentinel (Object.)]
  (fn [& args]
    (let [cached (.getOrDefault cache args sentinel)]
      (if (identical? sentinel cached)
         (comment compute the value)
         cached))
      
```